PennMobile/GSR-Booking/Controllers/GSRGroups/GSRGroupNewIntialController.swift

```diff
@@ -0,0 +1,289 @@
+
+
+    @objc func createGroupBtnAction(sender:UIButton!) {
+        //TODO: Consider adding appropriate error messages
+        guard let name = nameField.text else {return}
+        guard let color = colorLabel.text else {return}
+        
+        GSRGroupNetworkManager.instance.createGroup(name: name, color: color) { (success, groupID, errorMsg) in
+            if success {
+                DispatchQueue.main.async {
+                    let controller = GSRGroupInviteViewController()
+                    controller.groupID = groupID
+                    self.navigationController?.pushViewController(controller, animated: true)
+                }
+                self.delegate.fetchGroups()
+            }
+        }
+    }
+
+    @objc func cancelBtnAction(sender:UIButton!) {
+        self.delegate.fetchGroups()
+        dismiss(animated: true, completion:nil)
+    }
+}
+
+//Mark: Setup UI
+extension GSRGroupNewIntialController {
+    fileprivate func prepareUI() {
+        prepareCloseButton()
+        prepareNameField()
+        prepareGroupForLabel()
+        prepareSegmentedControl()
+        prepareColorLabel()
+        prepareColorCollection()
+        prepareCreateButton()
+    }
+}
+
+extension GSRGroupNewIntialController: UICollectionViewDelegate, UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 7
+    }
+
+    func collectionView(_ collectionView: UICollectionView,
+                        cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GSRColorCell.identifier, for: indexPath) as! GSRColorCell
+        let color = GSRGroup.parseColor(color: colorNames[indexPath.row]) ?? UIColor.blueLighter
+        cell.color = color
+        cell.borderColor = color.borderColor(multiplier: 1.5)
+        return cell
+    }
+
+    // MARK: - Collection View Delegate Methods
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let cell = collectionView.cellForItem(at: indexPath) as? GSRColorCell
+        cell?.toggleBorder()
+        createButton.isEnabled = true
+        if (nameChanged) {
+            createButton.backgroundColor = cell?.colorView.backgroundColor
+        } else {
+            createButton.backgroundColor = .grey4
+        }
+        chosenColor = cell?.colorView.backgroundColor
+        colorLabel.textColor = cell?.colorView.backgroundColor
+        colorLabel.text = colorNames[indexPath.item % colorNames.count]
+        colorLabel.font = UIFont.boldSystemFont(ofSize: 17)
+    }
+
+    // Deselect this time slot and all select ones that follow it
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        let cell = collectionView.cellForItem(at: indexPath) as? GSRColorCell
+        cell?.toggleBorder()
+    }
+}
+
+extension UIColor {
```

Comment:
This extension should be with the rest of the UIColor extensions, not in a GSR-related file